src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java

```diff
@@ -0,0 +1,66 @@
+package edu.kis.powp.jobs2d.drivers.decorator;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.CompoundCommand;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.features.DriverFeature;
+import edu.kis.powp.jobs2d.features.RecordingFeature;
+import edu.kis.powp.observer.Subscriber;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordingDriverDecorator implements Job2dDriver, Subscriber {
```

Comment:
SRP rule. Could you separate Subscriber from Job2dDriver?

Reply:
Moved Subscriber logic to separate class